src/algorithms/ctmrg.jl

```diff
@@ -140,18 +140,14 @@ function gauge_fix(envprev::CTMRGEnv{C,T}, envfinal::CTMRGEnv{C,T}) where {C,T}
             scalartype(T),
             MPSKit._lastspace(Tsfinal[end])' ← MPSKit._lastspace(M[end])',
         )
-
-        ρ_prev = transfermatrix_fixedpoint(Tsprev, M, ρinit)
-        ρ_final = transfermatrix_fixedpoint(Tsfinal, M, ρinit)
+        ρprev = transfermatrix_fixedpoint(Tsprev, M, ρinit)
+        ρfinal = transfermatrix_fixedpoint(Tsfinal, M, ρinit)
 
         # Decompose and multiply
-        Up, _, Vp = tsvd!(ρ_prev)
-        Uf, _, Vf = tsvd!(ρ_final)
-        Qprev = Up * Vp
-        Qfinal = Uf * Vf
-        σ = Qprev * Qfinal'
+        Qprev, = leftorth(ρprev)
+        Qfinal, = leftorth(ρfinal)
```

Comment:
we should probably restrict the compat for TensorKit to be above the version where the rrules for this got fixed, I think this is at least 0.12.4

Reply:
Definitely a good idea, just added that